src/pipeline.cc

```diff
@@ -250,6 +250,9 @@ class PipelineWorker : public Nan::AsyncWorker {
       if (shrink_on_load > 1) {
         // Reload input using shrink-on-load
         vips::VOption *option = VImage::option()->set("shrink", shrink_on_load);
+        if (baton->input->failOnError) {
```

Comment:
Similar question here, should the logic match? Perhaps move it into a helper? (I notice `accessMethod` is missing here - that looks like an existing bug though so I'll sort it later.)

Reply:
I'm not going to add a helper, but I'll push this change.